src/opm/parser/eclipse/Deck/DeckItem.cpp

```diff
@@ -410,15 +425,20 @@ bool DeckItem::equal(const DeckItem& other, bool cmp_default, bool cmp_numeric)
         break;
     case type_tag::fdouble:
         if (cmp_numeric) {
-            const std::vector<double>& this_data = this->dval;
-            const std::vector<double>& other_data = other.dval;
+            const auto& this_data = this->getData<double>();
+            const auto& other_data = this->getData<double>();
             for (size_t i=0; i < this_data.size(); i++) {
                 if (!double_equal( this_data[i] , other_data[i], rel_eps, abs_eps))
                     return false;
             }
         } else {
-            if (this->dval != other.dval)
-                return false;
+            if (this->raw_data == other.raw_data)
+                return (this->dval == other.dval);
+            else {
+                const auto& this_data = this->getData<double>();
```

Comment:
I do not understand this else branch: if `this->raw_date != other.raw_data` shouldn't we just return false?

Reply:
Perhaps put `if (this->raw_data != other.raw_data) return false` on the first line of the case, line 427? That way we also deal with the cmp_numeric case.

Reply:
If `this->raw_data != other.raw_data` that means that one member is in `raw` state and the other is in SI state. We just call `getData()`to ensure that both are `raw` before proceeding to actually compare. #mutable-taken-too-far

Reply:
Ok, it's fine.